Geometry_Engine/Query/IsNull.cs

```diff
@@ -0,0 +1,66 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.oM.Geometry;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Quantities.Attributes;
+using System.ComponentModel;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Checks if an IGeometry is null and outputs relevant error message.")]
+        [Input("geometry", "The geometry to test for null.")]
+        [Input("methodName", "The name of the method to reference in the error message.")]
+        [Output("pass", "A boolean which is true if the bar passes the null check.")]
+        public static bool IsNull(this IGeometry geometry, string methodName = "")
+        {
+            if (geometry == null)
+            {
+                //If the methodName is not provided, use StackTrace to get it, if the method was called indepedently use "Method".
+                if (string.IsNullOrEmpty(methodName))
+                {
+                    StackTrace st = new StackTrace();
+                    methodName = st.FrameCount > 0 ? st.GetFrame(1).GetMethod().Name : "Method";
```

Comment:
I think the method name should be cleaned up to given the user a nice readable message. Right now this generates:
![Screenshot 2021-05-17 163641](https://user-images.githubusercontent.com/6618854/118517059-cf589600-b72e-11eb-9d1e-8770f6f644b7.jpg)
But should read:
1. Cannot evaluate Integrate because geometry failed a null check.